scripts/check-lockfiles.mjs

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) 2023, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import path from 'node:path'
+import fs from 'node:fs'
+import { fileURLToPath } from 'node:url'
+
+const LOCKFILE_VERSION_TARGET = 2
+const guidance =
+    'To regenerate the lockfiles with the correct lockfile version, please run' +
+    ' `rm -rf package-lock.json node_modules/` in each impacted directory, and then use npm 7 or 8 to' +
+    ' run `npm install` in the monorepo root.'
+
+// __filename isn't available directly in ESM files
+const __filename = fileURLToPath(import.meta.url)
+const projectDir = path.join(__filename, '../..')
+const packagesDir = path.join(projectDir, 'packages')
+const dirs = fs
+    .readdirSync(packagesDir, {withFileTypes: true})
+    .filter((dirent) => dirent.isDirectory())
+    .map((dirent) => path.join(packagesDir, dirent.name))
+let exitCode = 0
+
+for (const dir of [/* monorepo root */ '.', ...dirs]) {
+    const filename = path.join(dir, 'package-lock.json')
+    // This file is executed as a postinstall script, so we can assume that the file will always exist
+    const lockfile = JSON.parse(fs.readFileSync(filename, 'utf8'))
+    if (lockfile.lockfileVersion !== LOCKFILE_VERSION_TARGET) {
+        console.error(
+            `Expected ${lockfile.name} to have lockfile version ${LOCKFILE_VERSION_TARGET}, but saw version ${lockfile.lockfileVersion}.`
```

Comment:
Are we locking in version 2? As far as I know, lockfile version 3 are compatible with lock file version 2, the only concern is when lock files is generated in version 1, which is from Node 14. When Node 14 is offically out of support, we should not have this issue i believe

Reply:
Just confirming what @alexvuong said: https://docs.npmjs.com/cli/v9/configuring-npm/package-lock-json#lockfileversion

Why do we want to lock in v2? To ensure compatibility with Node 14 / npm 6?

Reply:
We want to lock to _some_ version in order to ensure consistency across packages. Version 2 was an arbitrary choice because that's what we're already using.